checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -337,16 +335,34 @@ public void postAsMemberOf(AnnotatedTypeMirror type,
             AnnotatedTypeMirror owner, Element element) {
         super.postAsMemberOf(type, owner, element);
 
-        if (!HACK_DONT_CALL_POST_AS_MEMBER) {
-            if (element.getKind().isField()) {
-                Collection<? extends AnnotationMirror> declaredFieldAnnotations = getDeclAnnotations(element);
-                AnnotatedTypeMirror fieldAnnotations = getAnnotatedType(element);
-                computeFieldAccessType(type, declaredFieldAnnotations, owner,
-                        fieldAnnotations, element);
-            }
+        if (element.getKind().isField()) {
+            Collection<? extends AnnotationMirror> declaredFieldAnnotations = getDeclAnnotations(element);
+            AnnotatedTypeMirror fieldAnnotations = getAnnotatedType(element);
+            computeFieldAccessType(type, declaredFieldAnnotations, owner, fieldAnnotations, element);
         }
     }
 
+    /**
+     * Controls which hierarchies' qualifiers are changed based on the
+     * receiver type and the declared annotations for a field.
+     * (committed-only).
+     * @see #computeFieldAccessType
+     * @see #getAnnotatedTypeLhs(Tree)
+     */
+    private boolean computingAnnotatedTypeMirrorOfLHS = false;
+
+    @Override
+    public AnnotatedTypeMirror getAnnotatedTypeLhs(Tree lhsTree) {
+        boolean oldOnlyComputeFieldCommittedHeirachry = computingAnnotatedTypeMirrorOfLHS;
```

Comment:
I would name the local variable like the field, prepending "old". If nothing else, change Heirarchry -> Hierarchy.